client-app/src/Components/NewItemForm.tsx

```diff
@@ -15,36 +17,76 @@ interface FormErrors {
     [key: string]: string;
 }
 
+interface Option {
+    value: string;
+    label: string;
+    id?: number;
+}
+
 const NewItemForm: React.FC = () => {
     const [formData, setFormData] = useState<FormData>({
         itemType: '',
         currentStatus: 'Received',
         donorEmail: '',
+        donorId: null,
         program: '',
+        programId: null,
         imageUpload: [],
         dateDonated: '',
     });
 
-    const itemTypeOptions = [
-        { value: 'bicycle', label: 'Bicycle' },
-        { value: 'computer', label: 'Computer' },
-        // More item type options can be added here
+    const itemTypeOptions: Option[] = [
+        { value: 'Bicycle', label: 'Bicycle' },
+        { value: 'Computer', label: 'Computer' },
     ];
 
-    const donorEmailOptions = [
-        { value: 'email1', label: 'cooldude@gmail.com' },
-        { value: 'email2', label: 'cplusplushater@icloud.com' },
-        { value: 'email3', label: 'ISEBestBuilding@yahoo.com' },
-    ];
+    const [donorEmailOptions, setDonorEmailOptions] = useState<Option[]>([]);
+    const [programOptions, setProgramOptions] = useState<Option[]>([]);
+    const [previews, setPreviews] = useState<string[]>([]);
+    const [errors, setErrors] = useState<FormErrors>({});
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
+    const [successMessage, setSuccessMessage] = useState<string | null>(null);
 
-    const programOptions = [
-        { value: 'youthProgram', label: 'Youth Program' },
-        { value: 'retailSales', label: 'Retail Sales' },
-        { value: 'recycle', label: 'Recycle' },
-        { value: 'earnABicycle', label: 'Earn-a-bicycle' },
-        { value: 'earnAComputer', label: 'Earn-a-computer' },
-    ];
+    useEffect(() => {
+        const fetchDonorEmails = async () => {
+            try {
+                const response = await axios.get(
+                    `${process.env.REACT_APP_BACKEND_API_BASE_URL}/donor`,
+                );
+                const emailOptions = response.data.map((donor: any) => ({
+                    value: donor.firstName,
+                    label: donor.email,
+                    id: donor.id,
+                }));
 
+                setDonorEmailOptions(emailOptions);
+                console.log('Donor Options:', emailOptions);
+            } catch (error) {}
+        };
+
+        const fetchPrograms = async () => {
+            try {
+                const response = await axios.get(
+                    `${process.env.REACT_APP_BACKEND_API_BASE_URL}/program`,
```

Comment:
There should not be `'/' `before `program`. I am getting 404 error for this end point. Please update your .env file with this
`REACT_APP_BACKEND_API_BASE_URL="http://localhost:4000/"` it seems you did not use the leading '/' in your .env
Please stick to the existing .env variable

![image](https://github.com/user-attachments/assets/861e061b-870f-4d32-9429-bb5f622bb629)

The same applies to[ line 54](https://github.com/oss-slu/material-donor-mutual-assist/pull/50/files#diff-138ee185286f6c648bce397c4bbbad818dc0f04bb8e5cbe1b5b504858804dacdR54)